setup.py

```diff
@@ -175,6 +175,9 @@ def run(self):
                     # Temp fix for CI
                     # TODO: need a better way to determine generator
                     cmake_args.append('-DCMAKE_GENERATOR_PLATFORM=x64')
+                cmd_exists = lambda x: any(os.access(os.path.join(path, x), os.X_OK) for path in os.environ["PATH"].split(os.pathsep))
```

Comment:
Does `distutils.spawn.find_executable` work on windows?

Reply:
Yes, I'll try!

Reply:
Actually I would prefer to not auto opt in to use clang once it's available, instead, user can choose to use it by passing this extra flag via CMAKE_ARGS environment variable. Same for ninja. What do you think?

Reply:
@bddppq Okay then, I'll revert the changes and add the additional tests into the build matrix.